File: implementations/react/src/patterns/behavioral/state/new-state.ts
import type { TicketContext } from "./context";
import { InProgressState } from "./progress-state";
import { TicketState } from "./state";

export class NewState extends TicketState {
  private ctx: TicketContext;
  constructor(ctx: TicketContext) {
    super();
    this.ctx = ctx;
  }
  handleRequest() {
    this.ctx.commitDraft();
    this.ctx.setState(new InProgressState(this.ctx));
  }
}

File: implementations/flutter/design_patterns_flutter/lib/patterns/behavioral/state/new_state.dart
import 'package:design_patterns_flutter/patterns/behavioral/state/context.dart';
import 'package:design_patterns_flutter/patterns/behavioral/state/progress_state.dart';
import 'package:design_patterns_flutter/patterns/behavioral/state/state.dart';

class NewState extends TicketState {
  final TicketContext ctx;
  NewState(this.ctx);

  @override
  void handleRequest() {
    ctx.commitDraft();
    ctx.setState(InProgressState(ctx));
  }
}
